modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/request_controller.rb

```diff
@@ -33,28 +33,28 @@ def index
 
         def decide
           proc_id = form_attributes['procId']
+          ptcpnt_id = form_attributes['ptcpntId']
+          decision = normalize(form_attributes['decision'])
 
-          unless proc_id
-            raise ::Common::Exceptions::ParameterMissing.new('procId',
-                                                             detail: 'procId is required')
-          end
-
-          decision = form_attributes['decision']
-
-          unless decision && %w[accepted declined].include?(normalize(decision))
-            raise ::Common::Exceptions::ParameterMissing.new(
-              'decision',
-              detail: 'decision is required and must be either "ACCEPTED" or "DECLINED"'
-            )
-          end
+          validate_decide_params!(proc_id:, decision:)
 
           service = ManageRepresentativeService.new(external_uid: 'power_of_attorney_request_uid',
                                                     external_key: 'power_of_attorney_request_key')
 
+          poa_request = validate_ptcpnt_id!(ptcpnt_id:, proc_id:, service:) if decision == 'declined'
```

Comment:
what happens to line :57 if this fails? if we don't set the first / last name? does the error handler raise an error and prevent any 500's here?

Reply:
We _should_ [raise an exception](https://github.com/department-of-veterans-affairs/vets-api/blob/4bb506760e67791de1245bc32efaf81fa619f319/modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/request_controller.rb#L134) if there is no matching POA request. That said, if the matching POA request does not have a first name, we will still [send the email with an empty name](https://github.com/department-of-veterans-affairs/vets-api/blob/4bb506760e67791de1245bc32efaf81fa619f319/modules/claims_api/app/sidekiq/claims_api/va_notify_declined_job.rb#L68).